packages/core-data/src/selectors.js

```diff
@@ -153,6 +153,52 @@ export function getEntityRecords( state, kind, name, query ) {
 	return getQueriedItems( queriedState, query );
 }
 
+/**
+ * Returns a list of objects with each edited entity
+ * record and its corresponding edits.
+ *
+ * @param {Object} state State tree.
+ *
+ * @return {Array} The list of edited records with their edits.
+ */
+export const getEntityRecordChangesByRecord = createSelector(
```

Comment:
I'm not sure we'll need such a selector. Each time the returned value from a selector is that complex, I feel something is wrong. Can we just replace it with a selector that checks if an entity is Dirty?
I guess right now we'd also need a selector to retrieve all dirty records (hopefully we could remove that later).

Reply:
>I guess right now we'd also need a selector to retrieve all dirty records (hopefully we could remove that later).

Yes we would need that anyways.